amclient/amclient.py

```diff
@@ -571,6 +571,25 @@ def download_dip(self):
     def download_aip(self):
         return self.download_package(self.aip_uuid)
 
+    def delete_package(self, package_uuid):
+        """Create a deletion request for a package."""
+        params = {
+            "pipeline": self.pipeline_uuid,
+            "event_reason": self.event_reason,
+            "user_id": self.ss_user_id,
+            "user_email": self.ss_user_email,
```

Comment:
What do you think of passing all these `params` to the `delete_package` method explicitly like you've done with the `package_uuid` parameter? And then have a separate convenience method like `delete_aip` below that relies on the current state of the `AMClient` object?

I think in general it's a bit of a problem that we rely on initial (non-credentials) state of the `AMClient`. For example if you'd want to request deletion of two different packages by different users, and so on. You'd need separate `AMClient` objects or risk touching "some but not all" of the existing attributes in a single instance.

Reply:
Thank you for the suggestion, I'll do this! And I agree - it's easy to get yourself in trouble with `AMClient`'s state if you're not careful.